### PR TITLE
Skip AWS integration tests if credentials not configured.

### PR DESCRIPTION
**Pull request by jieru-hu**
<!-- Thank you for sending a PR and taking the time to improve Hydra -->

## Motivation

This is to address #1100 


### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan
1. CI
2. remove AWS CLI configurations on my local machine and run the tests
```
$ pytest -s tests/test_ray_aws_launcher.py 
================================================================================================================================ test session starts ================================================================================================================================
platform darwin -- Python 3.8.2, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
rootdir: /Users/jieru/workspace/hydra-fork/hydra/plugins/hydra_ray_launcher, inifile: pytest.ini
plugins: snail-0.1.0, hydra-core-1.0.3
collected 21 items                                                                                                                                                                                                                                                                  

tests/test_ray_aws_launcher.py::test_discovery PASSED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_1_job[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_2_jobs[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_not_sweeping_hydra_overrides[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_1_job_strict[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_1_job_strict_and_bad_key[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_2_optimizers[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_over_unspecified_mandatory_default[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_and_override[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncher::test_sweep_with_custom_resolver[ray_aws-overrides0-tmpdir0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_task_name[no_config-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_task_name[different_filename-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_task_name[different_filename_and_config_file_name_override-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_task_name[different_filename_and_config_file_name_override_and_command_line_override-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_sweeper_run_workdir[sweep_dir_config_override-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_sweeper_run_workdir[sweep_dir_cli_override-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_sweeper_run_workdir[sweep_dir_cli_overridding_config-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_sweeper_run_workdir[subdir:override_dirname-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_custom_sweeper_run_workdir[subdir:custom_override_dirname-tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_get_orig_dir_multirun[tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED
tests/test_ray_aws_launcher.py::TestRayAWSLauncherIntegration::test_to_absolute_path_multirun[tmpdir0-task_launcher_cfg0-extra_flags0] SKIPPED

=========================================================================================================================== 1 passed, 20 skipped in 2.62s ===========================================================================================================================
(pytest38) jieru-mbp:hydra_ray_launcher jieru$ 

```
## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: plugins/hydra_ray_launcher/integration_test_tools/create_integration_test_ami.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import tempfile
 import time
 from datetime import datetime
+from pathlib import Path
 
 import boto3
```
Comment (shagunsodhani) [inline plugins/hydra_ray_launcher/integration_test_tools/create_integration_test_ami.py:21]: Is `boto3` a dependency for ray (ie installed with ray) ?

Comment (jieru-hu) [reply plugins/hydra_ray_launcher/integration_test_tools/create_integration_test_ami.py:21]: I don't think so. But it is a dependency of this plugin.

**Comment by omry** [comment]
It will be easier if you explain what this is supposed to solve when we meet.

**Comment by jieru-hu** [comment]
> It will be easier if you explain what this is supposed to solve when we meet.

sure. I put some context in the motivation part, happy to chat more in person when we meet.

**Comment by jieru-hu** [comment]
@omry 
I've updated this PR per our chat last week (hardcode the env variables and skip tests if credentials are not set correctly)
Could you take a look when you get a chance? thanks!

File: plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py [hunk 2]
```diff
@@ -43,10 +45,23 @@
...
-instance_role = os.environ.get("INSTANCE_ROLE_ARN", "")
+
+aws_not_configured = True
+aws_not_configured_msg = "AWS credentials not configured correctly. Skipping AWS tests."
+try:
+    ec2 = boto3.client("ec2")
+    ec2.describe_regions()
+    aws_not_configured = False
+except (NoCredentialsError, NoRegionError):
+    pass
```
Comment (omry) [inline plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py:56]: Style wise, I would go:
```suggestion
try:
    ec2 = boto3.client("ec2")
    ec2.describe_regions()
    aws_not_configured = False
except (NoCredentialsError, NoRegionError):
    aws_not_configured = True
    aws_not_configured_msg = "AWS credentials not configured correctly. Skipping AWS tests."
```

**Review by omry — APPROVED** [review]
one nit, feel free to merge.

**Pull request merged**